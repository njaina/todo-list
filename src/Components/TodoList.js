import TodoItem from "./TodoItem";

const TodoList = ({ InputValues, handleCheck }) => (
    <div>
      {InputValues.map((todo, index) => (
        <TodoItem
          key={todo}
          todo={todo}
          handleCheck={handleCheck}
          index={index}
        />
      ))}
    </div>
  );
  export default TodoList;