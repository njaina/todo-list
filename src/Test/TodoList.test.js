import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../Components/TodoList';

describe('TodoList component', () => {
  it('renders correctly', () => {
    const handleCheck = jest.fn();
    const InputValues = ['item 1', 'item 2'];
    const tree = renderer
      .create(<TodoList InputValues={InputValues} handleCheck={handleCheck} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
