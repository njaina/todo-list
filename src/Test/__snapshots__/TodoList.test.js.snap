// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component renders correctly 1`] = `
<div>
  <div
    className="response"
  >
    <div
      className="element"
    >
      item 1
    </div>
    <div
      className="check"
    >
      <input
        onClick={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="response"
  >
    <div
      className="element"
    >
      item 2
    </div>
    <div
      className="check"
    >
      <input
        onClick={[Function]}
        type="checkbox"
      />
    </div>
  </div>
</div>
`;
